vendor/k8s.io/client-go/tools/cache/controller.go

```diff
@@ -354,6 +356,19 @@ func DeletionHandlingMetaNamespaceKeyFunc(obj interface{}) (string, error) {
 	return MetaNamespaceKeyFunc(obj)
 }
 
+// DeletionHandlingTypedKeyFunc checks for
+// DeletedFinalStateUnknownNamespacedName objects before calling
+// MetaNamespaceKeyFuncWithStrongType. It returns a
+// types.NamespacedName, which is a struct with Namespace and Name
+// fields, instead of a string key. This function is basically the same as
+// DeletionHandlingMetaNamespaceKeyFunc but for keys of type types.NamespacedName
+func DeletionHandlingTypedKeyFunc(obj interface{}) (types.NamespacedName, error) {
```

Comment:
You can't just edit the vendored dependencies.
Instead, you could add a similar func to `controllehelpers/cache.go` (doesn't exist now but feel free to add it).